### DISCUSSION TITLE
Prevent Root loader from re-fetching every time the page changes

### DISCUSSION THREAD

**Discussion by bohdanbirdie** [remix-run/remix#3779] [Q&A]
Hi!

I'm facing an issue of over fetching data from the root loader

The example Application pages structure I have is
```
/orders
/orders/:id
/create-order
```
I need a data from few endpoints to use on all the pages above, to be specific - I need to fetch enums, like Status, DeliveryTime and etc, and have them available everywhere. As well as the user profile.
Data loaded in the root Loader function and passed down via the outlet context.

The problem I got is that the Root Loader function is executed on every page transition client side, over fetching and hitting the API too often.

Here is the scenario (LF = Loader Function):
```
- GET /orders
- LF: load the data
---> API GET /api/status
---> API GET /api/delivery-time
---> API GET /api/me
---> API GET /api/orders
- LF: return the data [status, deliveryTime, me]
- Page is pre-rendered on the server and returned
- User navigates to /orders/1
-> LF: load the data
---> API GET /api/status !!!
---> API GET /api/delivery-time !!!
---> API GET /api/me !!!
---> API GET /api/orders/1
Loaders data is returned to the UI for the client-side rendering
```

I would like to avoid re-fetching the root loader data on every page change.
So far I tried adding `Cache-Control` headers to the root loader function, but that seems to be too not the best solution.

Are there any recipes for this kind of problem?

**Comment by jgb-solutions** [comment]
Sounds like you might want to cache your API responses somewhere. Redis or Workers KV if you are deploying to Cloudflare Workers. Remix re-fetches all the parent loaders by default. And root's loader is always run. So caching might be your best bet here.

**Reply by kiliman** [reply]
Remix only refetches active route loaders after a mutation (POST). This is called revalidation.

**Reply by jgb-solutions** [reply]
Yes that's right.

**Reply by bohdanbirdie** [reply]
Thanks, @jgb-solutions 
I was thinking about this option, but wanted to avoid added complexity

**Comment by kiliman** [comment]
By default, the root loader should only be refetched after mutations during validation.

If you're seeing it called on every request, it looks like you're doing a full page request for every navigation. To enable client-side routing make sure you have `<Scripts/>` in root and use `<Link>` instead of `<a>` for links.

**Reply by kiliman** [reply]
You can also opt out of reload, by using the `unstable_shouldReload` API.

https://remix.run/docs/en/v1/api/conventions#unstable_shouldreload

**Reply by bohdanbirdie** [reply]
I checked the for both Scripts and Link, all goo
But I have a Context Provider around the root Outlet, will play with that and see if it has any impact

Thanks for the tips!

**Reply by kiliman** [reply]
Without seeing actual code, I can only guess at typical causes. Good luck!
